packages/agreement-process/src/services/agreementActivationProcessor.ts

```diff
@@ -123,32 +130,67 @@ export async function createActivationEvent(
       updatedAgreementSeed
     );
 
-    return toCreateEventAgreementActivated(
-      {
-        ...updatedAgreement,
-        contract: apiAgreementDocumentToAgreementDocument(agreementContract),
-      },
-      agreement.metadata.version,
-      correlationId
-    );
-  } else {
-    if (authData.organizationId === agreement.data.producerId) {
-      return toCreateEventAgreementUnsuspendedByProducer(
-        updatedAgreement,
-        agreement.metadata.version,
-        correlationId
-      );
-    } else if (authData.organizationId === agreement.data.consumerId) {
-      return toCreateEventAgreementUnsuspendedByConsumer(
-        updatedAgreement,
+    return [
+      toCreateEventAgreementActivated(
+        {
+          ...updatedAgreement,
+          contract: apiAgreementDocumentToAgreementDocument(agreementContract),
+        },
         agreement.metadata.version,
         correlationId
-      );
-    } else {
-      throw genericError(
-        `Unexpected organizationId ${authData.organizationId} in activateAgreement`
-      );
-    }
+      ),
+    ];
+  } else {
+    return match([authData.organizationId, updatedAgreement.state])
```

Comment:
**Just minor opinion**
in other cases of activation process the updatedAgreement.state could be `SUSPENDED`,
so the activation flow produces AgreementUnsuspendedByProducer event type even when the output state is not `ACTIVE` but still remains or is updated to `SUSPENDED`, this behaviour could be not intuitive for reader, it's possible to add a bit comment to clarify that?